addons/mail/static/src/models/attachment/attachment.js

```diff
@@ -162,6 +162,46 @@ function factory(dependencies) {
             return clear();
         }
 
+        /**
+         * @private
+         * @returns {string|undefined}
+         */
+        _computeFormattedSize() {
+            if (!this.size) {
+                return clear();
+            }
+            const units = ["bytes", "Kb", "Mb", "Gb"];
+            let value = this.size;
+            let i = 0;
+            while (value >= 1000 && i < units.length) {
+                value /= 1000;
+                i++;
+            }
+            switch (units[i]) {
+                case "bytes" :
```

Comment:
I would not use such an array with technical string, which is only used for switch-case. This prone to typos.
Also the code is not correct in some cases, like if you deal with TeraBytes of data. Currently it crashes.

I suggest making this compute less algorithmic, so no `while` loop for instance.
Keep think plain and simple:
- "%s Bytes" for less than 1K
- "%s KB" for 1K-1M
- "%s MB" for 1M-1G
- "%s GB" for anything above 1G

Reply:
Okay I changed it